packages/@react-spectrum/searchwithin/src/SearchWithin.tsx

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import React from "react";
+import styles from "@adobe/spectrum-css-temp/components/searchwithin/vars.css";
+import {
+  classNames,
+  useDOMRef,
+  SlotProvider,
+  useStyleProps,
+  dimensionValue,
+} from "@react-spectrum/utils";
+
+import {
+  DOMRef,
+  DOMRefValue,
+  FocusableRefValue,
+  LabelPosition,
+} from "@react-types/shared";
+import { SpectrumSearchWithinProps } from "@react-types/searchfield";
+
+import { useProviderProps } from "@react-spectrum/provider";
+
+function SearchWithin(props: SpectrumSearchWithinProps, ref: DOMRef) {
+  props = useProviderProps(props);
+  let { styleProps } = useStyleProps(props);
+  let {
+    isQuiet,
+    isDisabled,
+    isRequired,
+    label,
+    children,
+    ...otherProps
+  } = props;
+
+  let domRef = useDOMRef(ref);
+
+  let pickerStyle = {
+    width: dimensionValue(100),
+  };
+
+  return (
+    <div
+      className={classNames(
+        styles,
+        "spectrum-SearchWithin",
+        styleProps.className
+      )}
+    >
+      <SlotProvider
+        slots={{
+          searchfield: {
+            UNSAFE_className: classNames(styles, "spectrum-SearchWithin-input"),
+          },
+          picker: {
+            placeholder: null,
+            UNSAFE_style: pickerStyle,
+            UNSAFE_className: classNames(
+              styles,
+              "spectrum-SearchWithin-picker"
+            ),
+            align: "end",
+            menuWidth: "size-6000",
```

Comment:
This one is a bit tricky to me, what I wanted was having the dropdown popover taking the full width of the SearchField and Picker, and even though it still doesn't look as expected (it's taking more width than I want), but if without this menuWidth param it will show right under the Picker like default, idk maybe I used it in a wrong way...

Reply:
Could you do something like ComboBox where you calculate the width of the searchfield and the picker and set the menu width appropriately: https://github.com/adobe/react-spectrum/blob/main/packages/@react-spectrum/combobox/src/ComboBox.tsx#L103-L110?